### ISSUE TITLE
Subfolders and Dovecot

### ISSUE THREAD

**Issue by reubenlillie** [nextcloud/mail#386]
Please pardon my confusion, @ChristophWurst and @oparoz, but has the Dovecot namespace prefixing issue (e.g., #185) been fixed via #271 or not? I know #185 was closed, but was it premature, or is my issue new? That is, does the "not working correctly" that I'm experiencing relate to what #271 attempted to fix, or is my issue, like #372, a separate kind of bug? The readme is still vague, and I'd like to know if it's something I'm doing wrong or a limitation for the current software.

In any case, I'm having difficulty similar to #372 on my instance (unless I'm also misunderstanding #372):

### Expected behaviour
"Show all folders" displays folders beyond inbox (e.g., Archive, Sent, Drafts, Junk, etc.)

### Actual behaviour
"Show all folders" displays no more folders, instead only "Collapse folders" toggle prompt.

### Mail app

**Mail app version:** 0.6.4

**Mailserver or service:** Dovecot

**Transport security - IMAP:** SSL 

**Transport security - SMTP:** SSL

**Number of accounts:** 1 (testing for development, ~250 accounts planned for client)

### Server configuration
**Operating system**: Linux server.***{DOMAIN}*** 2.6.32-696.el6.x86_64 #1 SMP Tue Mar 21 19:29:05 UTC 2017 x86_64

**Web server:** Apache (cgi-fcgi)

**Database:** mysql 5.6.35

**PHP version:** 7.1.5
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, readline, Reflection, SPL, session, standard, cgi-fcgi, bcmath, calendar, ctype, curl, dom, mbstring, ftp, gd, gettext, imap, json, exif, mcrypt, mysqlnd, PDO, Phar, posix, pspell, SimpleXML, soap, sockets, sqlite3, tidy, tokenizer, xml, xmlwriter, xsl, zip, mysqli, pdo_mysql, pdo_sqlite, wddx, xmlreader, xmlrpc

**Nextcloud version:** 12.0.0 - 12.0.0.29

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** https://nextcloud.com/install/

**Signing status:**
<details>
	<summary>Signing status</summary>

```
[]
```

</details>

**List of activated apps:**
<details>
	<summary>App list</summary>
```
Enabled:
 - activity: 2.5.2
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.3
 - comments: 1.2.0
 - contacts: 1.5.3
 - dav: 1.3.0
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_pdfviewer: 1.1.1
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - issuetemplate: 0.2.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.6.4
 - nextcloud_announcements: 1.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0

Disabled:
 - admin_audit
 - encryption
 - files_external
 - user_external
 - user_ldap
```
</details>

**The content of config/config.php:**
<details>
	<summary>Config report</summary>

```
{
    "instanceid": "ocn3td528q7q",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.***{DOMAIN}***"
    ],
    "datadirectory": "\/home\/***{DOMAIN}***\/public_html\/cloud\/data",
    "overwrite.cli.url": "https:\/\/cloud.***{DOMAIN}***",
    "dbtype": "mysql",
    "version": "12.0.0.29",
    "dbname": "***{DOMAIN}***_nextcloud",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_domain": "***{DOMAIN}***",
    "mail_smtpmode": "php",
    "mail_smtpauthtype": "LOGIN",
    "mail_from_address": "cloud",
    "app.mail.imaplog.enabled": true,
    "app.mail.smtplog.enabled": true,
    "app.mail.imap.timeout": 20,
    "app.mail.smtp.timeout": 2,
    "app.mail.transport": "php-mail"
}
```
</details>

**Are you using external storage, if yes which one:** files_external is disabled

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

### Client configuration
**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0

**Operating system:**

#### Web server error log (data/horde_imap.log)
<details>
	<summary>Web server error log</summary>
	```
	------------------------------
>> Wed, 31 May 2017 19:05:16 +0000
>> Connection to: imap://mail.***{DOMAIN}***.org:993/
>> Server connection took 0.0262 seconds.
S: * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE NAMESPACE AUTH=PLAIN AUTH=LOGIN] Dovecot ready.
C: 1 AUTHENTICATE PLAIN [INITIAL CLIENT RESPONSE (username: cloud@pghnaz.org)]
S: 1 OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE NAMESPACE NOTIFY SPECIAL-USE COMPRESS=DEFLATE QUOTA] Logged in
>> Command 1 took 0.0632 seconds.
C: 2 LIST () "" (*) RETURN (SPECIAL-USE)
S: * LIST () "." INBOX
S: * LIST (\UnMarked \Archive) "." INBOX.Archive
S: * LIST (\UnMarked \Junk) "." INBOX.spam
S: * LIST (\UnMarked \Junk) "." INBOX.Junk
S: * LIST (\UnMarked \Trash) "." INBOX.Trash
S: * LIST (\UnMarked \Drafts) "." INBOX.Drafts
S: * LIST (\UnMarked \Sent) "." INBOX.Sent
S: 2 OK List completed (0.001 + 0.000 + 0.001 secs).
>> Command 2 took 0.0018 seconds.
C: 3 STATUS INBOX.Junk (MESSAGES)
S: * STATUS INBOX.Junk (MESSAGES 0)
S: 3 OK Status completed (0.001 + 0.000 secs).
>> Command 3 took 0.0005 seconds.
C: 4 STATUS INBOX.Junk (MESSAGES)
S: * STATUS INBOX.Junk (MESSAGES 0)
S: 4 OK Status completed (0.001 + 0.000 secs).
>> Command 4 took 0.0003 seconds.
C: 5 STATUS INBOX.spam (MESSAGES)
S: * STATUS INBOX.spam (MESSAGES 0)
S: 5 OK Status completed (0.001 + 0.000 secs).
>> Command 5 took 0.0003 seconds.
C: 6 STATUS INBOX (MESSAGES RECENT UIDNEXT UIDVALIDITY UNSEEN)
S: * STATUS INBOX (MESSAGES 2 RECENT 0 UIDNEXT 3 UIDVALIDITY 1496251615 UNSEEN 0)
S: 6 OK Status completed (0.001 + 0.000 secs).
>> Command 6 took 0.0008 seconds.
C: 7 LOGOUT
S: * BYE Logging out
S: 7 OK Logout completed (0.001 + 0.000 secs).
>> Command 7 took 0.0009 seconds.
	```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
	<summary>Nextcloud log</summary>
	```
	empty
	```
</details>

#### Browser log
<details>
	<summary>Browser log</summary>
	```
	Insert your browser log here, this could for example include:

	a) The javascript console log
Content Security Policy: Directive ‘frame-src’ has been deprecated. Please use directive ‘child-src’ instead.  (unknown)
JQMIGRATE: Migrate is installed, version 1.4.0  core.js:7:542
Starting Mail …  mail.min.js:157:26485
initializing cache…  mail.min.js:152:21641
[ui] "content:loading" Array [  ]  mail.min.js:148:14867
[ui] "notification:request" Array [  ]  mail.min.js:148:14867
[account] "entities" Array [  ]  mail.min.js:148:14867
[ui] "sidebar:loading" Array [  ]  mail.min.js:148:14867
[ui] "search:set" Array [ "" ]  mail.min.js:148:14867
[ui] "content:loading" Array [ "Loading Inbox" ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[ui] "folder:changed" Array [  ]  mail.min.js:148:14867
[ui] "navigation:newmessage:show" Array [  ]  mail.min.js:148:14867
[ui] "sidebar:accounts" Array [  ]  mail.min.js:148:14867
[ui] "foldercontent:show" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ null ]  mail.min.js:148:14867
[message] "bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[message] "fetch:bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[ui] "message:loading" Array [  ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[message] "entity" Array [ Object, Object, 2 ]  mail.min.js:148:14867
[ui] "composer:events:undelegate" Array [  ]  mail.min.js:148:14867
[message] "flag" Array [ Object, Object, Object, "unseen", false ]  mail.min.js:148:14867
[ui] "messagesview:messageflag:set" Array [ 2, "unseen", false ]  mail.min.js:148:14867
[ui] "message:show" Array [ Object ]  mail.min.js:148:14867
[message] "load" Array [ Object, Object, Object ]  mail.min.js:148:14867
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead.  mail
[keyboard] "keyup" Array [ Object, 9 ]  mail.min.js:148:14867
[ui] "search:set" Array [ "" ]  mail.min.js:148:14867
[ui] "content:loading" Array [ "Loading Inbox" ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "foldercontent:show" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ null ]  mail.min.js:148:14867
[message] "bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[message] "fetch:bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[ui] "message:loading" Array [  ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[message] "entity" Array [ Object, Object, 2 ]  mail.min.js:148:14867
[ui] "composer:events:undelegate" Array [  ]  mail.min.js:148:14867
[message] "flag" Array [ Object, Object, Object, "unseen", false ]  mail.min.js:148:14867
[ui] "messagesview:messageflag:set" Array [ 2, "unseen", false ]  mail.min.js:148:14867
[ui] "message:show" Array [ Object ]  mail.min.js:148:14867
[message] "load" Array [ Object, Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "messagesview:messages:update" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[navigation] "folder" Array [ 3, "SU5CT1g=", false ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[notification] "favicon:change" Array [ "/apps/mail/img/favicon.png" ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 73 ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 65 ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 17 ]  mail.min.js:148:14867

	```
</details>

**Comment by reubenlillie** [comment]
@ChristophWurst, what if any info do you still need from me?

**Comment by ChristophWurst** [comment]
Hi @reubenlillie,

could you please try with a recent nightly build? PR #271 has not been released yet.

**Comment by ChristophWurst** [comment]
I've created a new build: https://github.com/nextcloud/mail/releases/tag/nightly-20170612 😉

**Comment by reubenlillie** [comment]
Thanks. I'll try to test within the next 48 hours.

**Comment by reubenlillie** [comment]
Here are the logs from my latest test using https://github.com/nextcloud/mail/releases/tag/nightly-20170612

`Undefined index: messages at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#163`
`Undefined index: messages at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#162`
`Undefined index: unseen at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#161`
`Creating account failed: Authentication failed.	2017-06-14T09:38:58-0500`
`Undefined index: messages at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#163`
`Undefined index: messages at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#162`
`Undefined index: unseen at ~/NEXTCLOUD_DIRECTORY/apps/mail/lib/Folder.php#161`
`Allowed memory size of 268435456 bytes exhausted (tried to allocate 92274385 bytes) at ~/NEXTCLOUD_DIRECTORY/lib/private/Security/Crypto.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL: Connection reset by peer at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL: Connection reset by peer at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL: Connection reset by peer at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL operation failed with code 1. OpenSSL Error messages: error:1409F07F:SSL routines:SSL3_WRITE_PENDING:bad write retry at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL: Connection reset by peer at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`
`fwrite(): SSL: Connection reset by peer at ~/NEXTCLOUD_DIRECTORY/apps/mail/vendor/pear-pear.horde.org/Horde_Smtp/Horde/Smtp/Connection.php#99`

Not sure if the because of the BlueHost/cPanel/Dovecot configuration or something else with the app codebase itself. I'm willing to experiment with guidance on my end. Please advise.

**Comment by japerezULT** [comment]
I'm experiencing this problem too, my Dovecot server follows the structure INBOX.sent, INBOX.trash, etc. But, even with the nightly build provided above, only the main one with received mail shows up. 

The logs don't show anything extraordinay apart from this:
`{"reqId":"UvVxGIls4dZFwIxZC7CC","level":3,"time":"2017-07-07T11:40:41+00:00","remoteAddr":"192.168.100.230","user":"ultreia","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/apps\/mail\/accounts\/1\/folders","message":"Undefined index: unseen at \/var\/www\/html\/nextcloud\/apps\/mail\/lib\/Folder.php#161","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko\/20100101 Firefox\/53.0","version":"12.0.0.29"}
{"reqId":"UvVxGIls4dZFwIxZC7CC","level":3,"time":"2017-07-07T11:40:41+00:00","remoteAddr":"192.168.100.230","user":"ultreia","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/apps\/mail\/accounts\/1\/folders","message":"Undefined index: messages at \/var\/www\/html\/nextcloud\/apps\/mail\/lib\/Folder.php#162","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko\/20100101 Firefox\/53.0","version":"12.0.0.29"}
{"reqId":"UvVxGIls4dZFwIxZC7CC","level":3,"time":"2017-07-07T11:40:41+00:00","remoteAddr":"192.168.100.230","user":"ultreia","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/apps\/mail\/accounts\/1\/folders","message":"Undefined index: messages at \/var\/www\/html\/nextcloud\/apps\/mail\/lib\/Folder.php#163","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko\/20100101 Firefox\/53.0","version":"12.0.0.29"}`

**Comment by yamlfd** [comment]
I have seen the documentation about the inbox prefixing, checked my server and this is disabled. However if I look into maildir folder the folders are prefixed. Not sure what dovecot /postfix really do now. In roundcube all folders and subfolders are there.

I only see the send / junk and 2 other folders, but not the full tree of inbox folders.

I installed the latest official version of mail app for next cloud. Looking forward to a fix to get rid of roundcube.

**Comment by reubenlillie** [comment]
Friendly update: I just tried the August 1 nightly build and the same issue persists.

Do we know if this issue is unique to Dovecot or if it is similar to the other open subfolder issues related to the "Show Folders"/"Collapse Folders" script?

Thanks for your help.

**Comment by douzepouze** [comment]
I can confirm the issue in the same configuration that @reubenlillie used.

**Comment by yamlfd** [comment]
This should also be the same for Courier IMAP. The INBOX prefix is typically set to prepare yourself for Courier and Dovecot compatibility.

**Comment by ftoledo** [comment]
I can confirm over Courier too.

**Comment by andreas-wolf** [comment]
Same problem with Cyrus IMAP. All I can see is a long list of Folders like INBOX.Archive.R... Furthermore the folder column is not resizeable making the whole app completely useless.

**Comment by winnieXY** [comment]
I can confirm this bug alsow with dovecot - with ~100 subfolders this looks like a complete mess and is not usable at all

**Comment by joe-speedboat** [comment]
Today I started with fresh dovecot installation and it drove mi crazy.
Folders are: INBOX.subfolder1.subfolder2
No way to fix from dovecot side.
Thunderbird and Nextcloud-Rainloop are working as expected.

NextCloud: 12.03 (stable)
OS: CentOS7 (up2date)
Dovecot: up2date
Tried everything you mentioned in Readme, it does not help at all.
Would be great to get that fixed.

**Comment by ChristophWurst** [comment]
Hey everyone,

could someone please do me a favor and test the fix in #652? For convenience, I've packed it into a tarball: 
[mail.tar.gz](https://github.com/nextcloud/mail/files/1526547/mail.tar.gz). Let me know if the folder hierarchy is correct with the patched app. Note that the UI will still show the prefix in the folder's name.

Btw, if anybody has a Docker image at hand that would allow us to test against one of the effected IMAP implementations, that would be great. I'd like to write an integration test for this and include it in our CI test suite.

**Comment by ChristophWurst** [comment]
> Btw, if anybody has a Docker image at hand that would allow us to test against one of the effected IMAP implementations, that would be great. I'd like to write an integration test for this and include it in our CI test suite.

Put differently: if anybody volunteers to search an appropriate Docker image from Docker hub, that would be awesome 😉

**Comment by ChristophWurst** [comment]
🏓 anybody got the chance to test my fix?

**Comment by ftoledo** [comment]
@ChristophWurst I could do the test, i have same behavior:

![pantallazo-2017-12-12 10-52-00](https://user-images.githubusercontent.com/95441/33888488-f7b07e12-df2b-11e7-847a-abd934edd2c0.png)

**Comment by ChristophWurst** [comment]
@ftoledo could you please enable debug mode and post your `data/horde_image.log` entries where the server lists your folders? Thanks.

**Comment by ChristophWurst** [comment]
Little reminder: I'm still waiting for someone to test my PR and give detailed feedback in case it does not work. If there's no or too little feedback from the community, I will close the PR and remove this ticket from my task and leave this to the community to fix.

**Comment by japerezULT** [comment]
@ChristophWurst I tried to test the patched version, however, if I overwrite the mail app folder with the new one, a "App upgrade required" screen pops up and re downloads the version from the store. If you could give some guidance around this I can try to test the folders problem.

**Comment by ChristophWurst** [comment]
@japerezULT I've created a tarball for you: https://github.com/nextcloud/mail/pull/652#issuecomment-352427619

**Comment by ftoledo** [comment]
@ChristophWurst im attach here the log (sorry i put them in #652  too)
[horde_imap.log](https://github.com/nextcloud/mail/files/1568331/horde_imap.log)

**Comment by japerezULT** [comment]
@ChristophWurst That tarball worked, thanks! However I had some unexpected behaviour that I would like to confirm if its intended or not:

The mail app was working fine with the older mail app I had in the server, however, after updating and overwriting, it stopped working and got an error, the culprit was a ".dovecot.sieve" file that was placed insice the user's Maildir, as soon as I removed that file it started working again. Don't know if this is the expected behaviour but at least with the older version it was working.

After that, all the Folders showed up, however I couldn't get the horde_imap.log to show up, correct me if I'm wrong, but adding:

`'app.mail.imaplog.enabled' => true,
  'app.mail.smtplog.enabled' => true,`

to config.php should suffice right?

**Comment by ChristophWurst** [comment]
> The mail app was working fine with the older mail app I had in the server, however, after updating and overwriting, it stopped working and got an error, the culprit was a ".dovecot.sieve" file that was placed insice the user's Maildir, as soon as I removed that file it started working again.

What version was installed before? Was it the latest from the app store. Actually, that error should be fixed now. If not, please file a separate ticket for that.



> After that, all the Folders showed up, however I couldn't get the horde_imap.log to show up, correct me if I'm wrong, but adding:

That used to be correct. Now we simply check [whether `'debug'` is set to `true`](https://github.com/nextcloud/mail/blob/9596a1f8c9e8db785eee2f7aa309a0a8ac58e181/lib/Account.php#L147).

**Comment by japerezULT** [comment]
@ChristophWurst 

I will recheck that and file a new ticket if I run into it again.

Regarding the subfolders here is my horde log,  there should me a mix of prefix and non prefix folders.

[horde_imap.log](https://github.com/nextcloud/mail/files/1572205/horde_imap.log)

**Comment by Ninos** [comment]
Sometimes inbox, trash, sent or draft-folder also have subfolders. Don't forget these default folders :-)

**Comment by ChristophWurst** [comment]
This is blocked by https://github.com/nextcloud/mail/issues/386#issuecomment-348917648. As long as I'm unable to locally reproduce the issues, fix them and have integration tests run on every build, I'm not continuing here.

So, again, if anybody has a working dovecot config that causes the issues above, please create a simple docker image with a single hard-coded user and let me know where to find it. I'll then fix the issue, write tests for it and integrate it into our CI builds. Thanks.

**Comment by j0s** [comment]
This is a reminder that pull request #728 is a temporary workaround to a problem that should no longer be needed once this has been resolved. When closing this bug, revert pull request #728.

**Comment by ChristophWurst** [comment]
New insights: horde detects the namespace capability
![bildschirmfoto von 2018-02-13 17-27-15](https://user-images.githubusercontent.com/1374172/36161214-67223d6a-10e3-11e8-82c8-fa124fc60fcc.png)

… and I have no clue how to handle those
![bildschirmfoto von 2018-02-13 17-27-58](https://user-images.githubusercontent.com/1374172/36161286-89950c4c-10e3-11e8-9e98-c3794a24d32a.png)

I probably cannot just blindly remove them.

**Comment by ChristophWurst** [comment]
@ftoledo I'm a bit confused. I'm using your test account (thanks a lot for providing it) but the folders are *not* prefixed. The only remaining issue is that all folders are shown as subfolder of the inbox.
![bildschirmfoto von 2018-02-13 17-47-38](https://user-images.githubusercontent.com/1374172/36162240-22d75746-10e6-11e8-9d91-9b799f122ddd.png)

**Comment by ChristophWurst** [comment]
Never mind, this was false positive. The folder names are only correct (non-prefixed) because the all have a special use set (inbox, sent, drafts, junk, …) and thus we translate the folder name automatically.

**Comment by ChristophWurst** [comment]
📖 https://tools.ietf.org/html/rfc2342

**Comment by ChristophWurst** [comment]
My observations:
* App works (at least no errors are thrown)
* Folders are shown as subfolders of the inbox (which other mail clients show that way too, but not all)
  * There's a bug with the collapse icon, it's hidden for the inbox
  * You can still click the inbox icon and subfolders will expand
* For some accounts, subfolder names include the parent folder prefix
  * Hence the INBOX.xxx names
  * They are removed on Gmail for some reason

**Comment by flesser** [comment]
I am running Mail 0.7.10 on Nextcloud 13.0.0 and still see this issue.
Folders on the root level are displayed correctly, but if I click on "expand", nested folders still show up as `Folder.sub-folder.third-level` etc. (As the actual maildir folders in the file system on the mailserver are named, but without the very first dot prefix). Thunderbird shows the folder structure correctly.

As for the Docker image you mentioned earlier in this thread:
I use https://github.com/hardware/mailserver without any modification of dovecot settings.
(Dovecot itself runs in the `mailserver` service defined in their [docker-compose.yml](https://github.com/hardware/mailserver/blob/master/docker-compose.sample.yml), you can skip the other stuff)

**Comment by ChristophWurst** [comment]
@flesser thanks for your feedback. Could you please file a new ticket?

Thanks a lot for the Docker info. It would be great if the Docker image is also somewhat runnable without Docker Compose.
